src/algorithm/boundingbox.rs

```diff
@@ -0,0 +1,209 @@
+use num::{Float};
+
+use types::{Bbox, Point, MultiPoint, LineString, MultiLineString, Polygon, MultiPolygon};
+
+/// Calculation of the bounding box of a geometry.
+
+pub trait BoundingBox<T: Float> {
+    /// Return a Bounding Box of a geometry
+    ///
+    /// ```
+    /// use geo::{Point, LineString, Coordinate};
+    /// use geo::algorithm::boundingbox::BoundingBox;
+    ///
+    /// let mut vec = Vec::new();
+    /// vec.push(Point::new(40.02f64, 116.34));
+    /// vec.push(Point::new(42.02f64, 116.34));
+    /// vec.push(Point::new(42.02f64, 118.34));
+    /// let linestring = LineString(vec);
+    /// let bbox = linestring.bbox().unwrap();
+    ///
+    /// println!("Bbox top left coordinates: {}, {}", bbox.xmin, bbox.ymax);
+    /// println!("Bbox bottom right coordinates: {}, {}", bbox.xmax, bbox.ymin);
+    /// ```
+    ///
+    fn bbox(&self) -> Option<Bbox<T>>;
+}
+
+fn get_bbox<T>(vect: &Vec<Point<T>>) -> Option<Bbox<T>>
+    where T: Float
+{
+    if vect.is_empty() {
+        return None;
+    }
+    if vect.len() == 1 {
+        return Some(Bbox{xmin: vect[0].x(), ymax: vect[0].y(),
+                         xmax: vect[0].x(), ymin: vect[0].y()})
+    } else {
+        let (mut xmax, mut xmin) = (T::neg_infinity(), T::infinity());
+        let (mut ymax, mut ymin) = (T::neg_infinity(), T::infinity());
+        for pnt in vect.iter() {
+            let (px, py) = (pnt.x(), pnt.y());
+            if px > xmax {
+                xmax = px;
+            } else if px < xmin {
+                xmin = px;
+            }
+            if py > ymax {
+                ymax = py;
+            } else if py < ymin {
+                ymin = py;
+            }
```

Comment:
On Mon, Jul 18, 2016 at 6:12 PM, Guillaume P. <notifications@github.com>
wrote:

> In src/algorithm/boundingbox.rs
> <https://github.com/georust/rust-geo/pull/41#discussion_r71177484>:
>
> benchmark on release mode? benchmark on debug is meaningless.
>
yes, I've created a bench file [0]. The point is that current
implementation of the fold function allocate several Bbox struct for each
point.
Should the bench file be in the repository as well?

[0] https://gist.github.com/zarch/3b03d3c5ac44f904f96aa699da383df9